libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -130,6 +130,10 @@ internal object MapboxNavigationTelemetry :
         logger: Logger?,
         locationsCollector: LocationsCollector = LocationsCollectorImpl(logger)
     ) {
+        reset()
+        needHandleReroute = false
+        needStartSession = false
+        sessionState = IDLE
```

Comment:
Calling `reset` at `initialize` should be enough to fix the flakiness but was wondering if we should reset the other state too (`needHandleReroute`, `needStartSession` and `sessionState`). Another thing that was wondering was if we should add `needHandleReroute`, `needStartSession` and `sessionState` as part of the `reset` which is also called from `sessionStop` 🤔

Reply:
I think we should reset them all